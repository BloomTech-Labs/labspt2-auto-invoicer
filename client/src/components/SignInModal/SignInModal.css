a {
  color: #000000;
  text-decoration: none;
}

.logins {
  align-items: center;
  display: flex;
  flex-direction: column;
  font-size: 1.6rem;
}

.google-login {
  align-items: center;
  border-radius: 2px;
  display: flex;
  height: 46px;
  width: 230px;
}

.facebook-login {
  align-items: center;
  border-radius: 2px;
  display: flex;
  height: 46px;
  width: 230px;
}

.facebook-login svg {
  margin-left: 12px;
}

.facebook-login span {
  margin-left: 11px;
}
