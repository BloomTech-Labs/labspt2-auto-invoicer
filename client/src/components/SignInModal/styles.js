export default theme => ({
  root: {
    flexGrow: 1
  },
  paper: {
    color: theme.palette.text.secondary,
    textAlign: 'center',
    width: 230,
    height: 46,
    borderRadius: 2,
    margin: '4px 0',
    boxShadow: '0 3px 6px rgba(0,0,0,0.20), 0 3px 6px rgba(0,0,0,0.20)'
  }
});
