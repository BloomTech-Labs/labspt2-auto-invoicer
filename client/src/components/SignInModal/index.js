import React from 'react';

import Modal from '../reusableComponents/Modal';
import { ReactComponent as GoogleLogo } from '../../assets/btn_google_light_normal_ios.svg';
import { ReactComponent as FacebookLogo } from '../../assets/f-ogo_RGB_HEX-58.svg';

import { withStyles } from '@material-ui/core/styles';
import Paper from '@material-ui/core/Paper';

import './SignInModal.css';

const styles = theme => ({
  paper: {
    color: theme.palette.text.secondary,
    textAlign: 'center',
    width: 230,
    height: 46,
    borderRadius: 2,
    margin: '4px 0',
    boxShadow: '0 3px 6px rgba(0,0,0,0.20), 0 3px 6px rgba(0,0,0,0.20)'
  }
});

const SignInModal = props => {
  const { classes } = props;
  return (
    <Modal close={props.click}>
      <div className="logins">
        <Paper elevation={1} className={classes.paper}>
          <a href={`${process.env.REACT_APP_BACKEND_URL}/auth/google`}>
            <div className="google-login">
              <GoogleLogo /> <span>Continue with Google</span>
            </div>
          </a>
        </Paper>
        <Paper elevation={1} className={classes.paper}>
          <a href={`${process.env.REACT_APP_BACKEND_URL}/auth/facebook`}>
            <div className="facebook-login">
              <FacebookLogo /> <span>Continue with Facebook</span>
            </div>
          </a>
        </Paper>
      </div>
    </Modal>
  );
};

export default withStyles(styles)(SignInModal);
