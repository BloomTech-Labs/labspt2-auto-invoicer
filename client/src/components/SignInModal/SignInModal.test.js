import React from "react";
import { render } from "react-testing-library";
import "jest-dom/extend-expect";
import SignInModal from "../SignInModal";

describe("<SignInModal />", () => {
  it("renders Sign In Modal", () => {
    const { asFragment } = render(<SignInModal />);
    expect(asFragment()).toMatchSnapshot();
  });
});
