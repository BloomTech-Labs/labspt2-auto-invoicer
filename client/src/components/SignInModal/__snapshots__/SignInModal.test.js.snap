// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignInModal /> renders Sign In Modal 1`] = `
<DocumentFragment>
  <div
    class="backdrop"
  >
    <div
      class="modal"
    >
      <span
        class="close-icon"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root-2"
          focusable="false"
          id="close"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
          />
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
        </svg>
      </span>
      <div
        class="logins"
      >
        <div
          class="MuiPaper-root-11 MuiPaper-elevation1-14 MuiPaper-rounded-12 SignInModal-paper-1"
        >
          <a
            href="http://localhost:5000/auth/google"
          >
            <div
              class="google-login"
            >
              <svg>
                btn_google_light_normal_ios.svg
              </svg>
               
              <span>
                Continue with Google
              </span>
            </div>
          </a>
        </div>
        <div
          class="MuiPaper-root-11 MuiPaper-elevation1-14 MuiPaper-rounded-12 SignInModal-paper-1"
        >
          <a
            href="http://localhost:5000/auth/facebook"
          >
            <div
              class="facebook-login"
            >
              <svg>
                f-ogo_RGB_HEX-58.svg
              </svg>
               
              <span>
                Continue with Facebook
              </span>
            </div>
          </a>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
